### ISSUE TITLE
VIP missing forever, after testing to call apache from shutted down node

### ISSUE THREAD

**Issue by dejoebad** [siderolabs/talos#9276]
- Talos version: 1.76
- Kubernetes version: 1.30.3
- Platform: AMD x64

i'm experiencing vip missing (removed). this how its produce:

1. using 2 CPs (192.168.11.100,192.168.11.101), using 1 vip 192.168.11.88
2. both running healthy and ready
3. deploy php apache
4. testing with 192.168.11.100:30004, works
5. testing with 192.168.11.101:30004, works
6. testing with 192.168.11.88:30004, works
7. shutdown node 192.168.11.101
8. testing with 192.168.11.100:30004, works
9. testing with 192.168.11.88:30004, works
10. testing with 192.168.11.101:30004, NOT working (due to shutdown)
11. at this point, node 192.168.11.100, missing its vip (vip is removed), controller-manager unhealthy, and never recovered since then.

to restore the condition, i have to reboot all. unfortunately, the process took at least 30 minutes

**Comment by smira** [comment]
If you use 2 controlplane nodes, your etcd cluster if of size 2 as well. Talos controlplane VIP uses etcd elections to decide which machine should announce it, so it depends on etcd being healthy.

etcd requires quorum of nodes to be available, for 2 nodes quorum is also 2 (`N/2 +1`). 

That's why _never_ use even number of etcd members, but rather odd (1, 3, 5, 7, etc.)

If you have etcd cluster of size 1, it tolerates failure of 0 nodes. Size 2 - 0 nodes. Size 3 - 1 node. Size 4 - 1 node. Size 5 - 2 nodes, etc.

**Comment by dejoebad** [comment]
> If you use 2 controlplane nodes, your etcd cluster if of size 2 as well. Talos controlplane VIP uses etcd elections to decide which machine should announce it, so it depends on etcd being healthy.
> 
> etcd requires quorum of nodes to be available, for 2 nodes quorum is also 2 (`N/2 +1`).
> 
> That's why _never_ use even number of etcd members, but rather odd (1, 3, 5, 7, etc.)
> 
> If you have etcd cluster of size 1, it tolerates failure of 0 nodes. Size 2 - 0 nodes. Size 3 - 1 node. Size 4 - 1 node. Size 5 - 2 nodes, etc.

I think what you mean is, at least 2 CPs should be on. Is that right?

**Comment by smira** [comment]
I mean something a bit more complicated which I posted above.

tl;dr don't use 2 controlplane nodes, use either 1 or 3.

If using 3 controlplane nodes, you controlplane will tolerate a single node failure

**Comment by dejoebad** [comment]
> I mean something a bit more complicated which I posted above.
> 
> tl;dr don't use 2 controlplane nodes, use either 1 or 3.
> 
> If using 3 controlplane nodes, you controlplane will tolerate a single node failure

yes, it worked. i've added 1 additional CP. now i have total 3 CPs. tries to shutdown 1 node (the one with vip). then quickly switch to another, and the rest 2 is runs healthy.

sorry, i don't speak with technical. am very new to this.
and thank you.
